private/protocol/rest/unmarshal.go

```diff
@@ -145,16 +146,20 @@ func unmarshalStatusCode(v reflect.Value, statusCode int) {
 	}
 }
 
-func unmarshalHeaderMap(r reflect.Value, headers http.Header, prefix string) error {
+func unmarshalHeaderMap(r reflect.Value, headers http.Header, prefix string, normalize bool) error {
```

Comment:
Unmarshal normalization makes sense. But does the normalization need to be applied to marshal as well? Since the wire protocol for HTTP header keys are uncased. Whereas with `normalize` we're saying `headers` http binding map keys are lowercase.